src/auction.js

```diff
@@ -224,25 +224,32 @@ function addBidToAuction(auctionInstance, bidResponse) {
 }
 
 // Video bids may fail if the cache is down, or there's trouble on the network.
-function tryAddVideoBid(auctionInstance, bidResponse, bidRequest, vastUrl) {
+function tryAddVideoBid(auctionInstance, bidResponse, bidRequest) {
+  let willAdd = false;
   if (config.getConfig('cache.url')) {
-    store([bidResponse], function(error, cacheIds) {
-      if (error) {
-        utils.logWarn(`Failed to save to the video cache: ${error}. Video bid must be discarded.`);
-
-        doCallbacksIfTimedout(auctionInstance, bidResponse);
-      } else {
-        bidResponse.videoCacheKey = cacheIds[0].uuid;
-        if (!vastUrl) {
-          bidResponse.vastUrl = getCacheUrl(bidResponse.videoCacheKey);
+    if (!bidResponse.videoCacheKey) {
+      willAdd = true;
+      store([bidResponse], function (error, cacheIds) {
+        if (error) {
+          utils.logWarn(`Failed to save to the video cache: ${error}. Video bid must be discarded.`);
+
+          doCallbacksIfTimedout(auctionInstance, bidResponse);
+        } else {
+          bidResponse.videoCacheKey = cacheIds[0].uuid;
+          if (!bidResponse.vastUrl) {
+            bidResponse.vastUrl = getCacheUrl(bidResponse.videoCacheKey);
+          }
+          // only set this prop after the bid has been cached to avoid early ending auction early in bidsBackAll
+          bidRequest.doneCbCallCount += 1;
+          addBidToAuction(auctionInstance, bidResponse);
+          auctionInstance.bidsBackAll();
         }
-        // only set this prop after the bid has been cached to avoid early ending auction early in bidsBackAll
-        bidRequest.doneCbCallCount += 1;
-        addBidToAuction(auctionInstance, bidResponse);
-        auctionInstance.bidsBackAll();
-      }
-    });
-  } else {
+      });
+    } else if (!bidResponse.vastUrl) {
+      bidResponse.vastUrl = getCacheUrl(bidResponse.videoCacheKey);
```

Comment:
We may not need line 249. When videoCacheKey exists, publisher is not using cache.url, publisher should use either Prebid Server cache with cacheMarkup (videoCacheKey = cache_id, vastUrl = cache_url) or its own cache server (videoCacheKey = imp_id or anything client side adapter wants, vastUrl = creative_depot_url or anything client side adapter wants).  And getCacheUrl method is using `cache.url` to build the Url.

Reply:
I know we're not using this line (because we turn vastUrl with videoCacheKey), but I'm not sure if other's might use it.  Is there no scenario where an adapter would return the videoCacheKey but expect that key to work against the url set with cache.url?

Reply:
And if that scenario is not possible, then this line should probably be substituted with some kind of errors or warnings.

Reply:
Agree, we can throw errors or warnings here. `videoCacheKey` and `vastUrl ` should be well defined in all cache scenarios.